Standardize SourceForge stable URL format

samford opened the pull request:
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Most of the `stable` URLs that use a `https://downloads.sourceforge.net` URL start with `https://downloads.sourceforge.net/project/`. However, there are around 30 that use a format like `https://downloads.sourceforge.net/aa-project/aalib-1.4rc5.tar.gz` which redirects to the `/project/` URL (e.g., `https://downloads.sourceforge.net/project/aa-project/aa-lib/1.4rc5/aalib-1.4rc5.tar.gz`) before redirecting again to a mirror.

This updates these URLs to use the appropriate `/project/` URL to avoid the first (non-mirror) redirection. For what it's worth, the only other `stable` SourceForge URL that doesn't use `https://downloads.sourceforge.net/project/` is `abook` (`http://abook.sourceforge.net/devel/abook-0.6.1.tar.gz`), since it's using a development version that isn't found in the SourceForge project files.

---

I've squashed this into one commit since this PR will simply be squash-and-merge'd.

Since these URL changes are using the same archive files (i.e., same location, same checksum, etc.), I'm going to cancel CI after tap_syntax completes. I've installed all of these from source locally and run `brew audit --strict --online` and `brew test` on each and the only failures are as follows.

## Audit failures

```
gauche:
  * C: 17: col 5: Formulae in homebrew/core (except e.g. cryptography, libraries) should not run build-time checks
gptsync:
  * C: 1: col 1: A `test do` test block should be added
libdv:
  * C: 1: col 1: A `test do` test block should be added
lxsplit:
  * C: 1: col 1: A `test do` test block should be added
msdl:
  * C: 1: col 1: A `test do` test block should be added
opencore-amr:
  * C: 1: col 1: A `test do` test block should be added
ttf2pt1:
  * C: 1: col 1: A `test do` test block should be added
Error: 7 problems in 7 formulae detected
```

## Test failures

The following `qrupdate` test failure occurs with the bottle as well, so it's not related to the `stable` URL update.

```
==> Testing qrupdate
==> gfortran -o test /usr/local/Cellar/qrupdate/1.1.2_13/share/qrupdate/tch1dn.f
==> ./test
Error: qrupdate: failed
An exception occurred within a child process:
  Test::Unit::AssertionFailedError: </PASSED\ \ \ 4\ \ \ \ \ FAILED\ \ \ 0/> was expected to be =~
<"\n" +
" testing Cholesky rank-1 downdate routines.\n" +
" All residual errors are expected to be small.\n" +
"\n" +
" sch1dn test:\n" +
"      residual error =              0.476837158203E-05       PASS\n" +
" dch1dn test:\n" +
"      residual error =              0.106581410364E-13       PASS\n" +
" cch1dn test:\n" +
"      residual error =              0.114611921163E-04       PASS\n" +
" zch1dn test:\n" +
"      residual error =              0.497379915032E-13       FAIL\n" +
"----------------------------------------------------------------------\n" +
" total:     PASSED   3     FAILED   1\n" +
"\n">.
```

dawidd6 commented:
I think you need to rebase, cause latest changes to ghc and lean-cli are somehow reverted in this PR.

samford commented:
Thanks for the heads up. Those `haskell-stack` and `lean-cli` reversions must have crept in when I rebased it before creating the PR. I rebased this and removed those changes, so it should be good now.
